lib/apis/v3/ippool.go

```diff
@@ -65,6 +66,22 @@ type IPPoolSpec struct {
 	NATOutgoingV1 bool `json:"nat-outgoing,omitempty" validate:"omitempty,mustBeFalse"`
 }
 
+// DoesMatchNode determines whether or not the IPPool's nodeSelector
+// matches the labels on the given node.
+func (pool IPPool) DoesMatchNode(n Node) (bool, error) {
```

Comment:
nit on naming: I'd call this "SelectsNode" rather than "MatchesNode", since "Selects" is the verb that selectors perform.